### PR TITLE
ci: use goma for windows and linux builds

### PR DESCRIPTION
**Pull request by jkleinsc**
#### Description of Change

This PR updates our Windows and Linux CI to use goma to speed up compilation.
#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -219,6 +219,15 @@ step-setup-env-for-build: &step-setup-env-for-build
     command: |
       # To find `gn` executable.
       echo 'export CHROMIUM_BUILDTOOLS_PATH="'"$PWD"'/src/buildtools"' >> $BASH_ENV
+      # Setup Goma if applicable
+      if [ ! -z "$RAW_GOMA_CONFIG" ] && [ "`uname`" == "Linux" ]; then
```
[the remark this answers is not in the document]
Comment (jkleinsc) [reply .circleci/config.yml]: @nornagon its a environment variable set on CircleCI.

[the remark this answers is not in the document]
Comment (nornagon) [reply .circleci/config.yml]: Is it just authentication info? Wherever possible I prefer to avoid CI relying on state that isn't in git.

**Pull request merged**